src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -3796,6 +3795,22 @@ theorem reverse_replicate {n : Nat} {x : BitVec w} :
     conv => lhs; simp only [replicate_succ']
     simp [reverse_append, ih]
 
+@[simp]
+theorem replicate_one {w : Nat} {x : BitVec w} (h : w = w * 1 := by rw [Nat.mul_one]) :
+    (x.replicate 1) = x.cast h := by
+  simp [replicate, h]
```

Comment:
I was a bit surprised this needed a cast, but [https://live.lean-lang.org/#codez=FAUwHghgtgDgNiABACjIgXIgchALgSg0TQCpEBGRAXmIxoCMBPRYRRAJwDM5hRJYEKNJhwEilMmhrCGzVh268AxAGMAFiBUBrbHgB0UAK5wgA](it seems) that indeed `w * 1` is not defeq to `w`, so this looks good